### PR TITLE
AOM-35: Implement drag and drop feature for the OWA upload section

### PR DESCRIPTION
**Pull request by mkiterian**
## JIRA TICKET NAME: 
[AOM-35 Implement drag and drop feature for the OWA upload section](https://issues.openmrs.org/browse/AOM-35)
### SUMMARY:
Currently, Drag and Drop feature is most wanted everywhere. We can achieve this task using most of the web technologies. We have already implemented this feature to System administration OWA, So better to introduce that feature to here also.

### CODE REVIEW

File: app/js/components/manageApps/ManageApps.jsx [hunk 1]
```diff
@@ -245,19 +242,28 @@ export default class ManageApps extends React.Component {
...
+              onClick={this.handleUpload}>
+              <span
+                className="glyphicon glyphicon-upload"
+              /> Upload
+            </button>
+            <button
+              id="clear-btn"
+              className="btn"
+              onClick={this.handleClear}
+            >
```
Comment (cnagadya) [inline app/js/components/manageApps/ManageApps.jsx:262]: Looks good though I've noticed some of the HTML element closing angle brackets like on line 261 are pushed to a new line. Is this intentional?

File: app/js/components/manageApps/AddAddon.jsx [hunk 2]
```diff
@@ -6,68 +6,64 @@
...
+    super();
+    this.state = {
+      disableClick: true,
+      dropzoneRef: {},
+    }
+    this.showMessage = this.showMessage.bind(this);
+  }
+
+  showMessage() {
+    if (this.props.files == null || this.props.files.length < 1) {
```
Comment (andela-dbamidele) [inline app/js/components/manageApps/AddAddon.jsx:24]: `===` instead of `==`

File: app/js/components/manageApps/ManageApps.jsx (hunk 1 above)
Comment (andela-dbamidele) [reply app/js/components/manageApps/ManageApps.jsx:262]: I think this is good. I do that always as well 😉

**Review by andela-dbamidele — COMMENTED** [review]
Great job! 👍

File: app/css/openmrs-addonmanager.css [hunk 3]
```diff
@@ -558,3 +561,45 @@ body.loading {
...
+    margin: auto;
+}
+
+.drop-zone-accept {
+    border: 5px solid #80ff80;
+}
+
+.drop-zone-reject {
+    border: 5px solid #ff8080;
+}
```
Comment (jomadoye) [inline app/css/openmrs-addonmanager.css]: Slight EOL error here

**Comment by jomadoye** [comment]
@cnagadya Awesome PR, good job!

**Comment by jomadoye** [comment]
@cnagadya I noticed the drag and drop field doesn't clear after successfully uploading an addon. I think it should be cleared.

File: app/js/components/manageApps/AddAddon.jsx [hunk 4]
```diff
@@ -6,68 +6,64 @@
...
+        <p>{this.props.files[0].name}</p>
+      );
+    }
+  }
+
+  render() {
+    return (
+      <div className="container-fluid">
+        <div className="row drop-zone-container">
+          <span>Select an OWA to upload</span>
```
Comment (jomadoye) [inline app/js/components/manageApps/AddAddon.jsx:45]: I think this span can be removed, we just need to change the wording on the file upload field to be more descriptive e.g `Drag addon file(s) here to upload or Click to here select`.
What do you think?

File: app/js/components/manageApps/ManageApps.jsx (hunk 1 above)
Comment (mkiterian) [reply app/js/components/manageApps/ManageApps.jsx:262]: @cnagadya  It was intentional. Helps me better visualize the code 😄 @andela-dbamidele , is it best practice to have it on the same line?

File: app/js/components/manageApps/AddAddon.jsx (hunk 2 above)
Comment (mkiterian) [reply app/js/components/manageApps/AddAddon.jsx:24]: The alternative is that a file is set i.e. props has a Filelist object. The initial value of files is null, set in ManageApps state and passed on to AddAddon via props. When a zip file is dragged and dropped, the files state is set to the Filelist object. When a non-zip file is drag and dropped the files state is set to an empty list

File: app/js/components/manageApps/AddAddon.jsx (hunk 4 above)
Comment (mkiterian) [reply app/js/components/manageApps/AddAddon.jsx:45]: @andela-jomadoye Thanks for pointing out the clearing of the drag and drop field on successful upload. I'll work on that.

On the span, I think the drag and drop field should have a title/label/description other than the instruction for a call to action. Maybe I should look for better wording?

**Comment by jacksono** [comment]
Well done.

**Pull request merged**